maven33-agent/src/main/java/jenkins/maven3/agent/Maven33Main.java

```diff
@@ -0,0 +1,215 @@
+        remoting.setParentRealm(launcher.getWorld().getRealm("plexus.core"));
+        remoting.addURL(remotingJar.toURI().toURL());
+
+        final Socket s = new Socket(agentIp,tcpPort);
+
+        Class remotingLauncher = remoting.loadClass("hudson.remoting.Launcher");
+        remotingLauncher.getMethod("main",
+                new Class[] { InputStream.class, OutputStream.class }).invoke(
+                null,
+                new Object[] {
+                        // do partial close, since socket.getInputStream and
+                        // getOutputStream doesn't do it by
+                        new BufferedInputStream(new FilterInputStream(s
+                                .getInputStream()) {
+                            public void close() throws IOException {
+                                s.shutdownInput();
+                            }
+                        }),
+                        new BufferedOutputStream(new RealFilter33OutputStream(s
+                                .getOutputStream()) {
+                            public void close() throws IOException {
+                                s.shutdownOutput();
+                            }
+                        }) });
+        System.exit(0);
+	}
+
+    /**
+     * Called by the code in remoting to add more plexus components.
+     * @since 1.3
+     */
+    public static void addPlexusComponents(URL[] modules) {
+        try {
+            ClassRealm realm = launcher.getWorld().getRealm("plexus.core");
+            for (int i=0; i<modules.length; i++) {
+                realm.addURL(modules[i]);
+            }
+        } catch (NoSuchRealmException e) {
+            throw new Error(e);
+        }
+    }
+
+    /**
+     * Called by the code in remoting to launch.
+     */
+    public static int launch( String[] args ) throws Exception {
+
+        try {
+            launcher.launch( args );
+        } catch ( Throwable e ) {
+            e.printStackTrace();
+            throw new Exception( e );
+        }
+        return launcher.getExitCode();
+    }
+
+    private static InputStream getClassWorldsConfStream() throws FileNotFoundException {
+        String classWorldsConfLocation = System.getProperty("classworlds.conf");
+        if (classWorldsConfLocation == null || classWorldsConfLocation.trim().length() == 0) {
+            classWorldsConfLocation = System.getenv("classworlds.conf");
+            if (classWorldsConfLocation == null || classWorldsConfLocation.trim().length() == 0) {
+                return Maven33Main.class.getResourceAsStream("classworlds.conf");
+            }
+        }
+        return new FileInputStream(new File(classWorldsConfLocation));
+    }
+	
+    /**
+     * Makes sure that this is Java5 or later.
+     */
+    private static void versionCheck() {
+        String v = System.getProperty("java.class.version");
+        if (v != null) {
+            try {
+                if (Float.parseFloat(v) < 49.0) {
```

Comment:
@olamy You are correct of course, I've modified it for Java 7 class version (51)